app/models/proposal.rb

```diff
@@ -45,48 +41,46 @@ def votes_in_tree
     # Rails.cache.fetch("/proposal/#{self.root.id}/votes_in_tree/#{updated_at}", :expires_at => 5.minutes) do
     # Code above seemed to never expire as of Rails 4
     # Proper cache should only cache when votes_in_tree > 100 or so
-    [self.root, self.root.descendants].flatten.map(&:votes_count).sum
+    all_related_proposals.map(&:votes_count).sum
     # end
   end
 
-  # def self_and_descendants    # dupe of all_related_proposals
-  #   [self.root, self.root.descendants].flatten
-  # end
-
   def related_proposals(related_sort_by = 'Most Votes')
-    all_proposals_in_tree = self.all_related_proposals
-    all_proposals_in_tree.delete(self.clone)
-    all_proposals_in_tree.select! { |proposal| proposal.has_support? }
+    other_supported_proposals = self.all_related_proposals.select(&:has_support?) - [self]
 
     case related_sort_by
     when "Most Votes"
-      all_proposals_in_tree.sort! { |p1, p2| p2.votes_count <=> p1.votes_count }
+      other_supported_proposals.sort! { |p1, p2| p2.votes_count <=> p1.votes_count }
     when "Least Votes"
-      all_proposals_in_tree.sort! { |p1, p2| p1.votes_count <=> p2.votes_count }
+      other_supported_proposals.sort! { |p1, p2| p1.votes_count <=> p2.votes_count }
     when "Most Recently Voted on"
-      all_proposals_in_tree.sort! { |p1, p2| p2.recent_vote <=> p1.recent_vote }
+      other_supported_proposals.sort! { |p1, p2| p2.recent_vote <=> p1.recent_vote }
     when "Oldest Most Recent Vote"
-      all_proposals_in_tree.sort! { |p1, p2| p1.recent_vote <=> p2.recent_vote }
+      other_supported_proposals.sort! { |p1, p2| p1.recent_vote <=> p2.recent_vote }
     else
-      all_proposals_in_tree.sort! { |p1, p2| p1.votes_count <=> p2.votes_count }
+      other_supported_proposals.sort! { |p1, p2| p1.votes_count <=> p2.votes_count }
     end
   end
 
+  def self.top_voted_proposal_in_tree
+    Proposal.roots.map{|root| root.all_related_proposals.max_by(&:votes_count) }.compact.uniq
+  end
+
   def all_related_proposals
-    [self.root, self.root.descendants].flatten
+    self.root.subtree
   end
 
   def supporting_statement
-    votes.where(user_id: self.user_id).first.comment
+    self.user.votes.first.comment
   end
 
   #TODO should be named "additional_supporting_votes?"
   def supporting_votes
-    votes.where("user_id != ?", self.user_id).order("created_at DESC")
+    votes.where.not(user_id: self.user_id).by_recency
   end
 
   def has_support?
-    self.votes_count != 0
+    self.votes_count > 0
   end
 
   def votes_percentage
```

Comment:
I usually find myself moving methods like this that return nicely formatted numbers into a Decorator.